src/formatDate.js

```diff
@@ -8,7 +8,68 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  let newFormat = '';
+  const fromSeparator = fromFormat[fromFormat.length - 1];
+  const toSeparator = toFormat[toFormat.length - 1];
+
+  let year = '';
+  let month = '';
+  let day = '';
+
+  let count = 0;
+
+  for (let i = 0; i < fromFormat.length - 1; i++) {
+    switch (fromFormat[i]) {
+      case 'YY':
+        year = date.slice(count, count + fromFormat[i].length);
+
+        if (+year < 30) {
+          year = 20 + year;
+        } else {
+          year = 19 + year;
+        }
+        break;
+
+      case 'YYYY':
+        year = date.slice(count, count + fromFormat[i].length);
+        break;
+
+      case 'DD':
+        day = date.slice(count, count + fromFormat[i].length);
+        break;
+
+      case 'MM':
+        month = date.slice(count, count + fromFormat[i].length);
+        break;
+    }
+    count += fromFormat[i].length + fromSeparator.length;
```

Comment:
The increment of `count` by `fromSeparator.length` is incorrect because the separator length is not always equal to the number of characters to skip in the `date` string. Instead, you should increment `count` by 1, which is the length of the separator in the `date` string.